src/main/java/com/dawn/jat/illuminati/post/controller/PostController.java

```diff
@@ -32,23 +32,25 @@
      */
     @GetMapping(value = "/{slug}")
     public ResponseEntity<Object> getPostBySlug(@PathVariable("slug") String slug) {
-        Optional<PostEntity> postEntity = postService.findBySlug(slug);
+        PostDto postDto = postService.getPost(slug);
 
-        if (!postEntity.isPresent()) {
+        if (Objects.isNull(postDto)) {
```

Comment:
it's alright to use null here, just note that the use of null sometimes not nice to have in the code and Optional is a option to deal with null.
https://medium.com/@elizarov/dealing-with-absence-of-value-307b80534903
There is actually a Null Object design pattern https://sourcemaking.com/design_patterns/null_object
:D 
In many cases, just try to avoid null as much as we can. For example, in this code 
```
if (Objects.isNull(map)) {
            return Collections.emptyList();
}
```
We return an empty list instead of null, which is nice to have, because we don't have to worry about null when calling this function.